### PR TITLE
Add Angular, Vue and React code snippets to dxTabPanel

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: concepts/05 Widgets/TabPanel/00 Overview.md [hunk 1]
```diff
@@ -25,6 +26,107 @@ The following code adds a simple **TabPanel** to your page.
         });
     });
 
+    <!--HTML-->
+    <div id='tabPanelContainer'></div>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/00 Overview.md]: Always use double quotes in the markup.

File: concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md [hunk 2]
```diff
@@ -1,139 +1,310 @@
...
+            );
+        }
+    }
 
-#####**Knockout**
+    export default App;
+
+---
+
+If you need a more flexible solution, define an [itemTemplate](/api-reference/10%20UI%20Widgets/dxTabPanel/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxTabPanel/Configuration/#itemTemplate') and [itemTitleTemplate](/api-reference/10%20UI%20Widgets/dxTabPanel/1%20Configuration/itemTitleTemplate.md '/Documentation/ApiReference/UI_Widgets/dxTabPanel/Configuration/#itemTitleTemplate') callback functions for views and tabs, respectively. In Angular and Vue, you can declare it in the markup. In React, you can use a rendering function (shown in the code below) or component
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md]:
```suggestion
If you need a more flexible solution, define [itemTemplate](/api-reference/10%20UI%20Widgets/dxTabPanel/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxTabPanel/Configuration/#itemTemplate') and [itemTitleTemplate](/api-reference/10%20UI%20Widgets/dxTabPanel/1%20Configuration/itemTitleTemplate.md '/Documentation/ApiReference/UI_Widgets/dxTabPanel/Configuration/#itemTitleTemplate') for views and tabs, respectively. In Angular and Vue, you can declare them in the markup. In React, you can use rendering functions (shown in the code below) or components.
```

File: concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md [hunk 3]
```diff
@@ -1,139 +1,310 @@
...
-        <div data-options="dxTemplate: { name: 'tab' } ">
-            <div style="margin:25px;">
-                <h1 data-bind="text: text"></h1>
-                <div style="text-align:left;" data-bind="foreach: dataArray">
-                    <p><span data-bind="text: propertyKey"></span>: <b data-bind="text: propertyValue"></b></p>
+        <div *dxTemplate="let content of 'item'">
+            <div style="margin: 25px;">
+                <h1>{{content.title}}</h1>
+                <div style="text-align: left;">
+					<p *ngFor="let key of getItemKeys(content.data)">
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md]: Please do not use tab symbols for indentation.

File: concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md [hunk 4]
```diff
@@ -1,139 +1,310 @@
...
+        },
+        data() {
+            return {
+                tabPanelItems: [{
+                    title: 'Info',
+                    data: { firstName: 'John', lastName: 'Smith', birthYear: 1986 }
+                }, {
+                    title: 'Contacts',
+                    data: { phone: '(555)555-5555', email: 'John.Smith@example.com' }
+                }];
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md]:
```suggestion
                }]
```

File: concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md [hunk 5]
```diff
@@ -1,139 +1,310 @@
...
+    export default App;
 
 ---
 
 #include common-demobutton with {
     url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/TabPanel/Overview/"
 }
 
-If you use jQuery alone, use <a href="http://api.jquery.com/category/manipulation/" target="_blank">DOM manipulation methods</a> to combine the HTML markup for items. To apply this markup, use the [itemTemplate](/api-reference/10%20UI%20Widgets/CollectionWidget/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxTabPanel/Configuration/#itemTemplate') and [itemTitleTemplate](/api-reference/10%20UI%20Widgets/dxTabPanel/1%20Configuration/itemTitleTemplate.md '/Documentation/ApiReference/UI_Widgets/dxTabPanel/Configuration/#itemTitleTemplate') callback functions for views and tabs, respectively.
+If you use jQuery, use <a href="http://api.jquery.com/category/manipulation/" target="_blank">DOM manipulation methods</a> to combine the HTML markup for tabpanel item. To apply this markup, use the **itemTemplate** and **itemTitleTemplate** callback functions as shown in the following code:
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/05 Customize Item Appearance.md]:
```suggestion
If you use jQuery, use <a href="http://api.jquery.com/category/manipulation/" target="_blank">DOM manipulation methods</a> to combine the HTML markup for **TabPanel** items. To apply this markup, use the **itemTemplate** and **itemTitleTemplate** callback functions as shown in the following code:
```

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 6]
```diff
@@ -1,44 +1,481 @@
...
+
+##### jQuery
+
     <!--JavaScript-->
-    $(function() {
-        $("#tabPanelContainer").dxTabPanel({
-            items: tabPanelItems,
+    $(function () {
+        $('#tabPanelContainer').dxTabPanel({
+            items: [{
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]: The data source doesn't add anything to the example. Let's remove it from all examples in this article.

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 7]
```diff
@@ -1,44 +1,481 @@
...
 
+    <!--HTML-->
+    <div id='tabPanelContainer'></div>
+
+##### Angular
+
+    <!--HTML-->
+    <dx-tab-panel
+        [items]="tabPanelItems"
+        [selectedIndex]="1">
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]: Let's two-way bind to the `selectedIndex` in Angular and Vue.

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 8]
```diff
@@ -1,44 +1,481 @@
...
+        title: 'Info',
+        text: 'This is Info Tab'
+    }, {
+        title: 'Contacts',
+        text: 'This is Contacts Tab'
+    }];
+
+    class App extends React.Component {
+        constructor() {
+            this.state = {
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md:163]:
```suggestion
            super(props);
            this.state = {
```

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 9]
```diff
@@ -1,44 +1,481 @@
...
+    })
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxTabPanel
+            :items="tabPanelItems"
+            :loop="false" />
+        </DxTabPanel>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]:
```suggestion

```

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 10]
```diff
@@ -1,44 +1,481 @@
...
+    })
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxTabPanel
+            :items="tabPanelItems"
+            :animation-enabled="true" />
+        </DxTabPanel>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]:
```suggestion

```

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 11]
```diff
@@ -1,44 +1,481 @@
...
+        text: 'This is Contacts Tab'
+    }];
+
+    class App extends React.Component {
+        render() {
+            return (
+                <TabPanel
+                    items={tabPanelItems}
+                    animationEnabled={true}
+                </TabPanel>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]:
```suggestion
                />
```

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 12]
```diff
@@ -1,44 +1,481 @@
...
+    })
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxTabPanel
+            :items="tabPanelItems"
+            :swipe-enabled="false"
+        </DxTabPanel>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]:
```suggestion
        />
```

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 13]
```diff
@@ -1,44 +1,481 @@
...
+        text: 'This is Contacts Tab'
+    }];
+
+    class App extends React.Component {
+        render() {
+            return (
+                <TabPanel
+                    items={tabPanelItems}
+                    swipeEnabled={false}
+                </TabPanel>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]:
```suggestion
                />
```

File: concepts/05 Widgets/TabPanel/15 Control the Behavior.md [hunk 14]
```diff
@@ -1,44 +1,481 @@
...
+        text: 'This is Contacts Tab'
+    }];
+
+    class App extends React.Component {
+        render() {
+            return (
+                <TabPanel
+                    items={tabPanelItems}
+                    loop={false}
+                </TabPanel>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/15 Control the Behavior.md]:
```suggestion
                />
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/TabPanel/00 Overview.md [hunk 15]
```diff
@@ -25,6 +26,107 @@ The following code adds a simple **TabPanel** to your page.
...
     });
 
+    <!--HTML-->
+    <div id="tabPanelContainer"></div>
+
+##### Angular
+
+    <!--HTML-->
+    <dx-tab-panel
+        [items]='tabPanelItems'>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/TabPanel/00 Overview.md]:
```suggestion
        [items]="tabPanelItems">
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]

**Pull request merged**